kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -200,10 +201,23 @@ protected CloseableIterator<FilteredColumnarBatch> getScanFiles(
 
   @Override
   public Row getScanState(Engine engine) {
+    // Check for metadata columns in the read schema and convert them to physical columns
+    List<StructField> convertedFields = new ArrayList<>();
+    for (StructField field : readSchema.fields()) {
+      Optional<StructField> converted =
+          MaterializedRowTrackingColumn.convertToPhysicalColumn(field, metadata);
+      if (converted.isPresent()) {
+        convertedFields.add(converted.get());
+      } else {
+        convertedFields.add(field);
+      }
+    }
+    StructType convertedSchema = new StructType(convertedFields);
+
```

Comment:
Move this whole logic into `convertToPhysicalColumn`

There are two approaches available, either:
1. we consider this logic of mapping to physical columns to be specific to row tracking, in which case it should be fully contained in `convertToPhysicalColumn`
2. We consider it to be generic and applying to possibly other metadata columns and convertToPhysicalColumn should be moved to ScanImpl.java and be implemented in a more generic manner, e.g. using a switch over known metadata column.

I think 1. is fine, we don't expect at this point to be adding new metadata column that map to physical columns